src/util/execute-lifecycle-script.js

```diff
@@ -24,6 +26,36 @@ export const IGNORE_MANIFEST_KEYS: Set<string> = new Set(['readme', 'notice', 'l
 // See https://github.com/yarnpkg/yarn/issues/2286.
 const IGNORE_CONFIG_KEYS = ['lastUpdateCheck'];
 
+async function getPnpParameters(config: Config): Promise<Array<string>> {
+  if (await fs.exists(`${config.lockfileFolder}/${constants.PNP_FILENAME}`)) {
```

Comment:
Maybe. If I remember correctly, the reason I simply check for the file to exist rather than the configuration is that someone might have used the `YARN_PLUGNPLAY_OVERRIDE` environment value to temporarily inhibe Plug'n'Play, in which case running `yarn run` won't work if they forget to add the override in the `yarn run` command as well.

Reply:
Makes sense. In that case I think it would be better to maybe check it once and ideally still move it to some other `pnpFileExists` config parameter or some such. Not overly critical though.